src/dimension.c

```diff
@@ -1355,11 +1355,12 @@ ts_dimension_add(PG_FUNCTION_ARGS)
 
 	if (!info.skip)
 	{
-		if (ts_hypertable_has_tuples(info.table_relid, AccessShareLock))
+		if (ts_hypertable_has_chunks(info.table_relid, AccessShareLock))
 			ereport(ERROR,
 					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 					 errmsg("hypertable \"%s\" is not empty", get_rel_name(info.table_relid)),
```

Comment:
"...is not empty" could be a confusing error message in case users deleted all data from the table already (but there are empty chunks). I suggest reformulating or keep this "has tuples" check in addition to the "has chunks" check.

Reply:
The errdetail actually gives more info about the error message. Also modified the error message .